neuralmonkey/tests/test_dataset.py

```diff
@@ -52,7 +53,8 @@ def test_glob(self):
                 s_data=[os.path.join(tmp_dir, "abc?"),
                         os.path.join(tmp_dir, "xyz*")])
 
-            self.assertEqual(dataset.get_series("data"), [["a"], ["b"], ["d"]])
+            series_iterator = dataset.get_series("data")
+            self.assertEqual(list(series_iterator), [["a"], ["b"], ["d"]])
 
```

Comment:
It would probably be good to also include following tests:

- check whether the batching is done correctly (ideally with more than one series)
- check whether the shuffling does not break things (this might be tricky)
- check whether the buffering works

Reply:
hotovo